### PR TITLE
Replace cpplint with clang-format

### PR DESCRIPTION
**Pull request by WillAyd**
cpplint isn't seemingly well maintained and focuses on C++, which we aren't using. I think clang-format is a much better tool for our formatting

### CODE REVIEW

**Comment by WillAyd** [comment]
Since we are adding a hook to pre-commit I'm guessing pre-commit.ci will have to be updated somehow to account for it. Does @MarcoGorelli or @lithomas1 know how that works?

**Comment by lithomas1** [comment]
> Since we are adding a hook to pre-commit I'm guessing pre-commit.ci will have to be updated somehow to account for it. Does @MarcoGorelli or @lithomas1 know how that works?

I think Marco's out on vacation right now, so I'll do my best to help.

Can we try not using pocc's wrapper around clang-format? I think that one doesn't install clang-format for you, so it won't work on pre-commit CI. 

You can try adding something manual like 
https://github.com/pandas-dev/pandas/blob/cc58350c92a9e24bef586cd1b898c01640b33abc/.pre-commit-config.yaml#L131-L388

(see the pyright one. 
We can try adding a new entry something like
```
id: clang-format
name: clang-format
entry: << clang-format command goes here >>
language: conda
additional_dependencies:
- clang-format
```
(I'm just typing this off of my head. Idea is to get pre-commit to pull clang-format off of conda. Don't expect it to work out of the box, though)

**Comment by WillAyd** [comment]
Ah OK thanks. I'll give that a shot - didn't realize pocc was missing something compared to the other repo to auto install

**Comment by WillAyd** [comment]
Any concerns on this one?

**Review by lithomas1 — APPROVED** [review]
Thanks Will,

This LGTM.

@MarcoGorelli any thoughts on this one?

**Review by MarcoGorelli — APPROVED** [review]
sorry for the delay, looks really good, nice one!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -166,6 +153,15 @@ repos:
         types: [pyi]
         args: [scripts/run_stubtest.py]
         stages: [manual]
+    -   id: clang-format
```
Comment (mroeschke) [inline .pre-commit-config.yaml:156]: Could you use https://github.com/pre-commit/mirrors-clang-format instead?

**Review by mroeschke — CHANGES_REQUESTED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @WillAyd

**Pull request merged**